scripts/getChangelog.py

```diff
@@ -1,90 +1,570 @@
+  data = json.loads(result)
+      
+  ## Parse response and save PR data
+  animation_state['text'] = "Parse PR metadata"
+  prs_for_csv = []
+
+  for n, pr_number in enumerate(unique_pr_numbers):
+      prdata = data['data']['repository'][f'pr{n}']
+      
+      # Create string with related issues
+      maximum_hit = False
+      related_issue_numbers = [relatedIssue['number'] for relatedIssue in prdata['closingIssuesReferences']['nodes']]
+      related_issues_string = ', '.join(map(str, related_issue_numbers))
+      if prdata['closingIssuesReferences']['totalCount'] > 10:
+        related_issues_string += " and more"
+      
+      # Create string with labels of the PR and its associated issues
+      unique_labels = set()
+      maximum_hit = False
+      if prdata['labels']['totalCount'] > 10:
+        maximum_hit = True
+      for label in prdata['labels']['nodes']:
+        unique_labels.add(label['name'])
+      for relatedIssue in prdata['closingIssuesReferences']['nodes']:
+        if relatedIssue['labels']['totalCount'] > 10:
+          maximum_hit = True
+        for label in relatedIssue['labels']['nodes']:
+          unique_labels.add(label['name'])
+          
+      unique_labels_list = list(unique_labels)
+      unique_labels_string = ', '.join(unique_labels_list)
+      if maximum_hit:
+        unique_labels_string += " and more (probably)"
+      
+      # Create string with issue & PR number(s) that need review replies
+      numbers = []
+      maximum_hit = False
+      if any(label['id'] == 'LA_kwDOAFAGHc8AAAABoGK6aw' for label in prdata['labels']['nodes']):
+        numbers.append(pr_number)
+      if prdata['closingIssuesReferences']['totalCount'] > 10 or prdata['labels']['totalCount'] > 10:
+        maximum_hit = True
+      for relatedIssue in prdata['closingIssuesReferences']['nodes']:
+        if any(label['id'] == 'LA_kwDOAFAGHc8AAAABoGK6aw' for label in relatedIssue['labels']['nodes']):
+          numbers.append(relatedIssue['number'])
+        if relatedIssue['labels']['totalCount'] > 10:
+          maximum_hit = True
+      numbers_str = ', '.join(map(str, numbers))
+      if maximum_hit:
+        numbers_str += " and more, possibly"
+      if numbers_str:
+        needs_review_reply_string = f"Yes ({numbers_str})"
+      else:
+        needs_review_reply_string = "No"
+      
+      # Store pr information in list
+      prs_for_csv.append({
+        'mergedAt': prdata['mergedAt'],
+        'URL': prdata['url'],
+        'title': prdata['title'],
+        'author': prdata['author']['login'],
+        'relatedIssues': related_issues_string,
+        'labels': unique_labels_string,
+        'needsReviewReplies': needs_review_reply_string,
+      })
+
+  # Create a list of dictionaries with commits for in CSV file
+      # /// NOTE maybe it's better to move this up to before the PR list generation (but after the list of unique PRs has been created) as it clears some of the memory used
```

Comment:
I don't think we need to care about memory here. To fill gigabytes of RAM on a PC, you need a lot of PRs.